src/sudo.c

```diff
@@ -557,7 +557,7 @@ get_user_info(struct user_details *ud)
     info[i] = sudo_new_key_val("user", pw->pw_name);
     if (info[i] == NULL)
 	goto oom;
-    ud->username = info[i] + sizeof("user=") - 1;
+    ud->username = pw->pw_name;
```

Comment:
This is not safe since `pw` points to storage that will be overwritten by the next call to getpwnam() or getpwuid().  The return value from sudo_new_key_val() is dynamically allocated which is why the the code uses it instead.